src/runtime/parcelset/parcelhandler.cpp

```diff
@@ -369,33 +339,15 @@ namespace hpx { namespace parcelset
         return result;
     }
 
-    ///////////////////////////////////////////////////////////////////////////
-    namespace detail
-    {
-        // The original parcel-sent handler is wrapped to keep the parcel alive
-        // until after the data has been reliably sent (which is needed for zero
-        // copy serialization).
-        void parcel_sent_handler(boost::system::error_code const& ec,
-            parcelhandler::write_handler_type const& f, parcel const& p)
-        {
-            // invoke the original handler
-            f(ec, p);
-
-            // inform termination detection of a sent message
-            if (!p.does_termination_detection())
-                hpx::detail::dijkstra_make_black();
```

Comment:
It's not needed. The senders in the parcelport implementations need to keep the parcel alive anyhow in order to call the sent handler which has the signature
`void(boost::system::error_code const& ec, parcel const & p)`;